### PR TITLE
JS -- Fix doc.getField and add missing field methods

### PR DESCRIPTION
**Pull request by calixteman**
- getField("foo") was wrongly returning a field named "foobar";
 - field object had few missing unimplemented methods

### CODE REVIEW

File: test/unit/scripting_spec.js [hunk 1]
```diff
@@ -159,6 +159,64 @@ describe("Scripting", function () {
         done.fail(ex);
       }
     });
+
+    it("should get field using a path", async function (done) {
```
Comment (brendandahl) [inline test/unit/scripting_spec.js:163]: Why async and done?

File: src/scripting_api/doc.js [hunk 2]
```diff
@@ -828,15 +828,50 @@ class Doc extends PDFObject {
       return searchedField;
     }
 
+    const parts = cName.split("#");
```
Comment (brendandahl) [inline src/scripting_api/doc.js:831]: It looks like fields should be a tree according to the spec. Why not store the fields as a tree and walk it to find the field?

**Review by brendandahl — CHANGES_REQUESTED** [review]

File: test/unit/scripting_spec.js (hunk 1 above)
Comment (calixteman) [reply test/unit/scripting_spec.js:163]: Yeah it's useless.
Let's fixing that in a follow-up.

Comment (brendandahl) [reply test/unit/scripting_spec.js:163]: Is there a reason this needs to be done in a follow up? Seems like a good time to fix it here.

Comment (calixteman) [reply test/unit/scripting_spec.js]: Done here: https://github.com/mozilla/pdf.js/pull/12830/commits/ccef734ebb80e50b0ed15ddc0f4c26cc1619f1ad

**Pull request merged**